### PR TITLE
Add support for account migration to Liberapay

### PR DESCRIPTION
**Pull request by Changaco**
Per <https://github.com/gratipay/inside.gratipay.com/issues/1196#issuecomment-336618468>. As of now this PR only adds hidden infrastructure, it doesn't touch the UI. I don't think I should be the one who writes the page explaining Gratipay's shutdown.

The new migration endpoint uses normal participant authentication as well as a shared secret which provides proof that the request is coming from Liberapay.

### CODE REVIEW

**Comment by mattbk** [comment]
!m @Changaco

**Comment by Changaco** [comment]
The Liberapay side still needs some work on the UI. I'll finish it tomorrow.

**Comment by Changaco** [comment]
Here it is: https://github.com/liberapay/liberapay.com/pull/773

**Comment by Changaco** [comment]
This is ready for review and merge, as is the Liberapay side.

File: defaults.env [hunk 1]
```diff
@@ -108,3 +108,6 @@ PROJECT_REVIEW_TOKEN=
 RAISE_SIGNIN_NOTIFICATIONS=no
 
 LOAD_BRAINTREE_FORM_ON_HOMEPAGE=no
+
+LIBERAPAY_BASE_URL=http://localhost:8339
+LIBERAPAY_SECRET=fake
```
Comment (mattbk) [inline defaults.env]: Do I need something here to test? Or do I need Liberapay running locally too?

Comment (Changaco) [reply defaults.env]: If you want to test the full migration process then yes, you need both Gratipay and Liberapay running locally.

Comment (mattbk) [reply defaults.env]: Thanks! Working on it...

**Comment by mattbk** [comment]
I'm still getting
```
400
Bad Request
Missing key: u'secret'
```
at /about/me/migrate. Running both Gratipay (this branch) and Liberapay (master) locally on default ports.

**Comment by Changaco** [comment]
You have to add `?secret=fake` to the URL. (You don't actually need Liberapay running for this part.)

**Comment by mattbk** [comment]
Okay, so that gives me a JSON of my various account data.

**Comment by mattbk** [comment]
Does the form template in `migrate.html` get used?

**Comment by Changaco** [comment]
@mattbk No. As I said in the original comment this PR is only hidden infrastructure.

**Comment by Changaco** [comment]
I've deployed account migration on Liberapay, to whom and how should I send the real shared secret for the `LIBERAPAY_SECRET` environment variable?

**Comment by mattbk** [comment]
Thanks @Changaco for your patience.

**Comment by whit537** [comment]
> I've deployed account migration on Liberapay, to whom and how should I send the real shared secret for the `LIBERAPAY_SECRET` environment variable?

I guess that would be me. Is email secure enough, do you think?

**Comment by whit537** [comment]
> Does the form template in `migrate.html` get used?

@mattbk Once this PR is landed we can utilize that template on the homepage in #4679.

**Comment by Changaco** [comment]
> I guess that would be me. Is email secure enough, do you think?

Yes. I've just sent it.

**Comment by Changaco** [comment]
Before showing off the migration button it would be a good idea to test that it actually works, all the way to charging a card in USD if possible. Any volunteers, or should I do it?

**Comment by rohitpaulk** [comment]
I did a test run... Looks like I hit a 500.

screen shot 2017-11-02 at 9 40 51 am

**Comment by rohitpaulk** [comment]
I like the clear explanation of what can/cannot be imported. :+1:

**Comment by Changaco** [comment]
> I did a test run... Looks like I hit a 500.

Should be fixed now.

**Pull request merged**